File: GremsReliquary/src/main/java/gremsReliquary/relics/normal/TimeIsMoney.java
package gremsReliquary.relics.normal;

import com.badlogic.gdx.graphics.Texture;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.vfx.GainPennyEffect;
import gremsReliquary.GremsReliquary;
import gremsReliquary.relics.AbstractGremRelic;
import gremsReliquary.util.TextureLoader;

import static com.megacrit.cardcrawl.core.CardCrawlGame.sound;

public class TimeIsMoney extends AbstractGremRelic {
    public static final String ID = GremsReliquary.makeID(TimeIsMoney.class.getSimpleName());
    public static final Texture IMG = TextureLoader.getTexture("gremsReliquaryResources/images/relics/BrokenMirror.png");
    public static final Texture OUTLINE = TextureLoader.getTexture("gremsReliquaryResources/images/relics/outline/BrokenMirror.png");
    public static int amount = 2;
    
    private boolean usedThisTurn = false;
    
    public TimeIsMoney() {
        super(ID, IMG, OUTLINE, RelicTier.UNCOMMON, GremRelicType.NORMAL, LandingSound.CLINK);
    }
    
    public void atTurnStart() {
        AbstractCreature source = AbstractDungeon.player;
        flash();
        sound.play("GOLD_JINGLE");
        AbstractDungeon.player.gainGold(amount);
        for (int i = 0; i < amount; ++i) {
            AbstractDungeon.effectList.add(new GainPennyEffect(source, this.currentX, this.currentY, source.drawX, source.drawY, true));
        }
    }
    
    // Description
    @Override
    public String getUpdatedDescription() {
        return DESCRIPTIONS[0] + amount + DESCRIPTIONS[1];
    }
}

File: GremsReliquary/src/main/kotlin/kotlinReliquary/relics/normal/TimeIsMoney.kt
package kotlinReliquary.relics.normal

import com.megacrit.cardcrawl.core.CardCrawlGame.sound
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.relics.AbstractRelic
import com.megacrit.cardcrawl.vfx.GainPennyEffect
import gremsReliquary.GremsReliquary
import gremsReliquary.util.TextureLoader
import kotlinReliquary.relics.AbstractGremRelic

class TimeIsMoney : AbstractGremRelic(ID, IMG, OUTLINE, AbstractRelic.RelicTier.COMMON, AbstractGremRelic.RelicType.NORMAL, AbstractRelic.LandingSound.CLINK) {

    companion object {
        val ID = GremsReliquary.makeID(TimeIsMoney::class.java.simpleName)
        val IMG = TextureLoader.getTexture("gremsReliquaryResources/images/relics/TimeIsMoney.png")
        val OUTLINE = TextureLoader.getTexture("gremsReliquaryResources/images/relics/outline/TimeIsMoney.png")
        var amount = 2
    }

    override fun atTurnStart() {
        val source = AbstractDungeon.player
        flash()
        sound.play("GOLD_JINGLE")
        AbstractDungeon.player.gainGold(amount)
        for (i in 0 until amount) {
            AbstractDungeon.effectList.add(GainPennyEffect(source, this.currentX, this.currentY, source.drawX, source.drawY, true))
        }
    }

    override fun getUpdatedDescription(): String {
        return DESCRIPTIONS[0] + amount + DESCRIPTIONS[1]
    }
}